### ISSUE TITLE
Custom Module Objective Message Duplication

### ISSUE THREAD

**Issue by Dylan-Buchanan** [PikaMug/Quests#2153]
### What is your Quests version / build number (do *not* say "latest")?

4.8.3

### Describe the problem and how to replicate it.

When given the objective for a quest that uses a custom quest objective the message is duplicated.

### Any console errors to share?

```shell
No console errors but I assume it is an error with the Quest API
```

### If applicable, can you provide an example quest from quests.yml?

```yaml
ordered:
        '1':
          custom-objectives:
            custom1:
              name: Total Blocks Broken Objective
              count: 1
              data:
                Total: '150'
                Obj Name: This is doubled and shouldn't be
          start-event: a-mine-pickaxe
          delay: 2
          start-message: '&aMiner: &fLet''s start with the basics'
```

**Comment by PikaMug** [comment]
What caused your assumption that the Quests API is at fault? Have you reached out to the module developer?

**Comment by Dylan-Buchanan** [comment]
My assumption is due to the fact that any built-in objectives do not have this duplication issue but any custom module whether it's another developer or the modules I made myself have this issue. I'd need to go back and check for the specific version but I'd like to note that reverting to previous versions of quests does solve the issue so it may very well not be Quests API (which is not very helpful for bug fixing I know). And regarding reaching out, as I said even my own modules don't work. Here is a simple example I used to check this to recreate the duplication issue on a fresh module:

package me.sir_draco;

import me.blackvein.quests.CustomObjective;
import me.blackvein.quests.Quest;
import me.blackvein.quests.Quester;
import me.blackvein.quests.Quests;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;

import java.util.Map;

public class BreakBlockObjective extends CustomObjective implements Listener {
    private static Quests quests = (Quests) Bukkit.getServer().getPluginManager().getPlugin("Quests");

    public AnyBreakBlockObjective() {
        setName("Break Blocks Objective");
        setAuthor("Sir_Draco");
        setShowCount(true);
        setCountPrompt("Set how many blocks to break");
        addStringPrompt("Total", "Set how many blocks to break", 1);
        setDisplay("Break %Total% Blocks (%count%)");
    }


    @EventHandler
    public void onBlockBreak(BlockBreakEvent event) {
        Player p = event.getPlayer();
        Quester qp = quests.getQuester(p.getUniqueId());
        for (Quest q : qp.getCurrentQuests().keySet()) {
            int index = -1;
            for (int i = 0; i < qp.getCurrentStage(q).getCustomObjectives().size(); i++) {
                if (qp.getCurrentStage(q).getCustomObjectives().get(i).getName().equals(this.getName())) {
                    index = i;
                    break;
                }
            }

            Map<String, Object> m = getDataForPlayer(p, this, q);

            if (index > -1) {
                int total = Integer.parseInt((String) m.get("Total"));
                int count = qp.getQuestData(q).getCustomObjectiveCounts().getFirst();
                if (count < total) {
                    incrementObjective(p, this, 1, q);
                }
                return;
            }
        }
    }
}

**Comment by Dylan-Buchanan** [comment]
I just tested and confirmed the bug does not exist using version 4.7.2 but does with version 4.8.x

**Comment by PikaMug** [comment]
Thanks for the clarification. The built-in objectives essentially use a different mechanism than custom ones, so this isn't too surprising. Now, it's possible I may have fixed this in 5.0.0-rc.1, so that's what I'll be using to proceed with testing. Note that the documentation hasn't been updated yet, but I'll keep you abreast of this as well.

**Comment by PikaMug** [comment]
So I wasn't able to replicate this on 5.0.0-rc.1 using any of my modules nor your example. Could be an issue with older versions. I've updated your code for 5.x here, although it doesn't work as expected because you're using `addStringPrompt` as a secondary count prompt, which isn't what it's intended for.

```
package me.sir_draco;

import me.pikamug.quests.BukkitQuestsPlugin;
import me.pikamug.quests.module.BukkitCustomObjective;
import me.pikamug.quests.player.Quester;
import me.pikamug.quests.quests.Quest;
import org.bukkit.Bukkit;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockBreakEvent;

import java.util.Map;
import java.util.UUID;

public class BreakBlockObjective extends BukkitCustomObjective implements Listener {
    private static final BukkitQuestsPlugin quests = (BukkitQuestsPlugin) Bukkit.getServer().getPluginManager().getPlugin("Quests");

    public BreakBlockObjective() {
        setName("Break Blocks Objective");
        setAuthor("Sir_Draco");
        setItem("DIAMOND_PICKAXE", (short)0);
        setShowCount(true);
        setCountPrompt("Set how many blocks to break");
        addStringPrompt("Total", "Set how many blocks to break", "1");
        setDisplay("Break %Total% Blocks (%count%)");
    }


    @EventHandler
    public void onBlockBreak(BlockBreakEvent event) {
        if (quests == null) {
            return;
        }
        UUID uuid = event.getPlayer().getUniqueId();
        Quester qp = quests.getQuester(uuid);
        for (Quest q : qp.getCurrentQuests().keySet()) {
            int index = -1;
            for (int i = 0; i < qp.getCurrentStage(q).getCustomObjectives().size(); i++) {
                if (qp.getCurrentStage(q).getCustomObjectives().get(i).getName().equals(this.getName())) {
                    index = i;
                    break;
                }
            }

            Map<String, Object> m = getDataForPlayer(uuid, this, q);

            if (index > -1) {
                int total = Integer.parseInt(m.get("Total").toString());
                int count = qp.getQuestDataOrDefault(q).getCustomObjectiveCounts().getFirst();
                if (count < total) {
                    incrementObjective(uuid, this, q, 1);
                }
                return;
            }
        }
    }
}
```
